app/celery/process_delivery_status_result_tasks.py

```diff
@@ -265,3 +276,243 @@ def sms_status_update(
     except Exception:
         current_app.logger.exception('Failed to check_and_queue_callback_task for notification: %s', notification.id)
         statsd_client.incr(f'clients.sms.{sms_status.provider}.status_update.error')
+
+
+def can_retry_sms_request(
+    status: str,
+    retries: int,
+    max_retries: int,
+    sent_at: datetime,
+    retry_window: datetime.timedelta,
+) -> bool:
+    """Determine if a retry is allowed.
+
+    Determine if a retry is allowed based on the status, number of retries,
+    the maximum retries allowed, the time the message was sent, and
+    the retry window.
+
+    Parameters:
+        status (str): Current notification status.
+        retries (int): The number of retries already attempted.
+        max_retries (int): The maximum number of retries allowed.
+        sent_at (datetime): The timestamp when the message was initially sent.
+        retry_window (timedelta): The allowable time window for retries.
+
+    Returns:
+        bool: True if retrying is allowed, False otherwise.
+    """
+    # Calculate the time elapsed since the message was sent
+    time_elapsed = datetime.datetime.utcnow() - sent_at
+
+    return (
+        (retries is not None)
+        and (status == NOTIFICATION_SENDING)
+        and (retries <= max_retries)
+        and (time_elapsed < retry_window)
+    )
+
+
+def get_sms_retry_delay(retry_count: int) -> int:
+    """Calculate the retry delay for SMS delivery with random jitter.
+
+    Delays were chosen to cover transient errors and small scale service disruptions
+
+    Note: Delays limited to under 900s due to AWS SQS limitation
+
+    Assumes up to 2 retries, repeating final delay for subsequent:
+    1st retry: 60 seconds +/- 10%
+    2nd retry: 10 minutes (600 seconds) +/- 10%
+
+    Parameters:
+        retry_count (int): The retry attempt number.
+
+    Returns:
+        int: Delay in seconds with applied jitter.
+    """
+    delay_with_jitter_by_retry_count = {
+        1: (60, 6),  # 60 seconds +/- 6 seconds (10%)
+        2: (600, 60),  # 10 minutes +/- 60 seconds (10%)
+    }
+
+    # Safeguard against retry counts outside the defined range
+    # Default to largest delay
+    default_delay_with_jitter = (600, 60)
+
+    base_delay, jitter_range = delay_with_jitter_by_retry_count.get(retry_count, default_delay_with_jitter)
+
+    # Apply jitter
+    delay = int(base_delay + random.randint(-jitter_range, jitter_range))  # nosec non-cryptographic use case
+
+    return delay
+
+
+def update_sms_retry_count(notification_retry_id: str, initial_value: int = 0, ttl: int | None = None) -> int:
```

Comment:
suggest adding a comma after the last argument so that the args are broken out on separate lines (easier to read).